include/configs/sc589-mini.h

```diff
@@ -241,11 +242,14 @@
 #define CONFIG_SYS_SPL_ARGS_ADDR	0xcf8cc000 /* dtb + args load address(SDRAM) */
 
 #define CONFIG_SYS_SPI_U_BOOT_OFFS  0x12000
-#define CONFIG_SYS_SPI_ARGS_OFFS	0x92000 /* CONFIG_SYS_SPI_U_BOOT_OFFS + 0x80000 */
-#define CONFIG_SYS_SPI_ARGS_SIZE	0xE000	/* 56KiB */
-#define CONFIG_SYS_SPI_KERNEL_OFFS	0xa0000 /* CONFIG_SYS_SPI_ARGS_OFFS + CONFIG_SYS_SPI_ARGS_SIZE */
-#define CONFIG_SYS_SPI_KERNEL_SIZE	0x1460000 /* 20MB */
-#define CONFIG_SYS_SPI_INITRAM_OFFS	0x1500000 /* CONFIG_SYS_SPI_KERNEL_OFFS + CONFIG_SYS_SPI_KERNEL_SIZE */
+#define CONFIG_SYS_SPI_U_BOOT_SIZE  0x80000    /* 512KiB */
+#define CONFIG_SYS_SPI_DTB_OFFS		0x92000    /* CONFIG_SYS_SPI_U_BOOT_OFFS + CONFIG_SYS_SPI_U_BOOT_SIZE */
+#define CONFIG_SYS_SPI_DTB_SIZE		0x40000    /* 256KiB */
+#define CONFIG_SYS_SPI_ARGS_OFFS	0xD2000    /* CONFIG_SYS_SPI_DTB_OFFS + CONFIG_SYS_SPI_DTB_SIZE */
+#define CONFIG_SYS_SPI_ARGS_SIZE	0x500000  /* 5MB */
+#define CONFIG_SYS_SPI_KERNEL_OFFS	0x5D2000  /* CONFIG_SYS_SPI_ARGS_OFFS + CONFIG_SYS_SPI_ARGS_SIZE */
+#define CONFIG_SYS_SPI_KERNEL_SIZE	0x1400000  /* 20MB */
+#define CONFIG_SYS_SPI_INITRAM_OFFS	0x19D2000  /* CONFIG_SYS_SPI_KERNEL_OFFS + CONFIG_SYS_SPI_KERNEL_SIZE */
```

Comment:
What are the current size of the uboot, dtb, args, kernel and initram sizes for the actual binaries, compared to the space we are allocating? How much growth space is there in each of these areas?

Reply:
Current size:
u-boot-spl.ldr: 24KB
u-boot.bin: 272KB
uImage: 2.3MB
dtb: 19KB
initram: 2.8MB
args: 5MB reserved

Reply:
I think the declared sizes are good, the current demo is significantly stripped back. These sizes should mean that users don't need to resize when their app grows. But we need to be aware that we are reserving a lot of memory.
Let's make sure that this is documented on the wiki.